Automation/S3/delete_s3_buckets.py

```diff
@@ -0,0 +1,194 @@
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError, NoCredentialsError
+from colorama import init, Fore
+import argparse
+
+COL_SIZE = 15
+init(autoreset=True)
+
+parser = argparse.ArgumentParser(description="Deletes S3 buckets.")
+parser.add_argument('--AccessKeyId', dest='aws_access_key_id', required=True)
+parser.add_argument('--SecretAccessKey', dest='aws_secret_access_key', required=True)
+parser.add_argument('--RegionName', dest='region_name', default='us-west-2', help='Region Name (default: us-west-2)')
+
+args = parser.parse_args()
+
+def main(aws_access_key_id, aws_secret_access_key, region_name):
+    '''
+    The main function for the S3 bucket deletion script
+    '''
+
+    print(Fore.GREEN + "Starting S3 bucket deletion script...")
+    try:
+        client_config = Config(
+            region_name = region_name
+        )
+
+        s3_client = boto3.client(
+            's3',
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            config=client_config
+        )
+
+        print("Retrieving S3 bucket list...")
+        list_buckets_response = s3_client.list_buckets()
+        print(Fore.GREEN + "Bucket list retrieved! Cycling through list...\n")
+
+        buckets_to_delete = []
+        for bucket in list_buckets_response['Buckets']:
+            bucket_name = bucket['Name']
+            
+            # Retrieve Bucket Information
+            print(f'Retrieving bucket: {bucket_name} information...')
+            bucket_owner_email = get_owner_email(s3_client, bucket_name)
+            bucket_object_list = get_bucket_object_list(s3_client, bucket_name)
+
+            # Display that information
+            print(Fore.GREEN + f'{"Bucket:":<{COL_SIZE}}{bucket_name}')
+            print(Fore.GREEN + f'{"OwnerEmail:":<{COL_SIZE}}{bucket_owner_email}')
+            bucket_object_list_prepared_for_deletion = []
+            if len(bucket_object_list) != 0:
+                print(Fore.GREEN + "Bucket Objects:")
+                for bucket_object in bucket_object_list:
+                    # Prepare just in case we want to delete the bucket
+                    bucket_object_list_prepared_for_deletion.append({
+                        'Key': bucket_object
+                    })
+                    print(bucket_object)
```

Comment:
This is an odd `print` statement. It will be helpful to add some context. What is the purpose of it?

Reply:
This is so that the user can check out what objects are in the bucket before choosing to delete. I can comment it, but is that useful as far as making that decision goes?